packages/core/src/exchange/approveCurrency.ts

```diff
@@ -0,0 +1,32 @@
+import type { TransactionResponse } from '@ethersproject/abstract-provider'
+import type { BigNumber, Signer } from 'ethers'
+import type { Sdk } from '../graphql'
+import type { Address, ChainId } from '../types'
+import { toAddress, toCurrencyId } from '../utils/convert'
+import { sendTransaction } from '../utils/transaction'
+
+type CurrencyApproval = {
+  chain: ChainId
+  currency: Address
+  amount: BigNumber
```

Comment:
Should use only BigNumberish?
```suggestion
  amount: BigNumberish
```

Reply:
No but I will create one 👍

Reply:
[d023835](https://github.com/liteflow-labs/liteflow-js/pull/142/commits/d023835d92ca8f1877c8ddd09ab1c1cff0b7f67a)